### PR TITLE
Use webp over png and lazy load images

### PR DESCRIPTION
**Pull request by vincanger**
## Description

Some minor image performance boosts

## Contributor Checklist

> Make sure to do the following steps if they are applicable to your PR:

- [ ] **Update e2e tests**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/template/e2e-tests](/template/e2e-tests) also.
- [x] **Update demo app**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/opensaas-sh/app_diff](/opensaas-sh/app_diff) also. Check [/opensaas-sh/README.md](/opensaas-sh/README.md) for details.
- [x] **Update docs**: If needed, update the [/opensaas-sh/blog/src/content/docs](/opensaas-sh/blog/src/content/docs).

### CODE REVIEW

File: template/app/src/client/components/AppNavBar.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const navigation = [
   { name: 'Blog', href: BlogUrl },
 ];
 
-const NavLogo = () => <img className='h-8 w-8' src={logo} alt='Your SaaS App' />;
+const NavLogo = () => <img className='h-8 w-8' src={logo} decoding='async' alt='Your SaaS App' />;
```
Comment (Martinsos) [inline template/app/src/client/components/AppNavBar.tsx:22]: Interesting, article here claims there isn't much improvement normally https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decoding , which suggests one should normally not bother with setting this value.

How have you decided it is worth setting it?

File: template/app/src/landing-page/components/Testimonials.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function Testimonials({ testimonials }: { testimonials: Testimoni
                 </blockquote>
                 <figcaption className='mt-6 text-base text-white'>
                   <a href={testimonial.socialUrl} className='flex items-center gap-x-2'>
-                    <img src={testimonial.avatarSrc} className='h-12 w-12 rounded-full' />
+                    <img src={testimonial.avatarSrc} loading='lazy' decoding='async' className='h-12 w-12 rounded-full' />
```
Comment (Martinsos) [inline template/app/src/landing-page/components/Testimonials.tsx:25]: I did a quick read about `loading` here: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/loading .

It says that in Firefox it has to go before `src` to take effect. Silly requirement, but maybe let's make sure that happens? Maybe React does it for us, but likely it doesn't, and better not to rely on it.

Also, I am guessing we shouldn't be setting `loading='lazy'` for every image we have? Or should we? What is the strategy about it, how do we make decision what is loaded lazily and what eagerly? I read that one con of lazy loading is that things around image might need to move around if height and width are not known upfront.

**Review by Martinsos — APPROVED** [review]
Approved, but ask some questions!

File: template/app/src/landing-page/components/Testimonials.tsx (hunk 2 above)
Comment (vincanger) [reply template/app/src/landing-page/components/Testimonials.tsx:25]: I thought that for everything below the main fold, lazy should be fine. I also tested it on the main hero image, which is kind of below the fold, and I don't see any noticeable difference.

File: template/app/src/client/components/AppNavBar.tsx (hunk 1 above)
Comment (vincanger) [reply template/app/src/client/components/AppNavBar.tsx:22]: Yeah, you're right, after reading up on it it seems like it's more likely designed for swapping out a thumbnail with a large image once it's done loading, something we're not doing here

File: template/app/src/landing-page/components/Testimonials.tsx (hunk 2 above)
Comment (Martinsos) [reply template/app/src/landing-page/components/Testimonials.tsx:25]: Ok soudns good!

File: template/app/src/client/components/AppNavBar.tsx (hunk 1 above)
Comment (Martinsos) [reply template/app/src/client/components/AppNavBar.tsx:22]: Ok! So we remove this one/

**Pull request merged**